### PR TITLE
refactor: add an interface for checking if the bucket is empty

### PR DESCRIPTION
**Pull request by Smityz**
I add an interface to check if the bucket is empty, so we don't need to use `get_bucket(1)` to do these things.
BTW, I add a group of unit tests to ensure the rate is always > 0

Ref: https://github.com/apache/incubator-pegasus/pull/829#discussion_r740053407

### CODE REVIEW

**Review by Shuo-Jia — APPROVED** [review]

File: src/utils/test/token_bucket_throttling_controller_test.cpp [hunk 1]
```diff
@@ -109,7 +110,8 @@ class token_bucket_throttling_controller_test : public ::testing::Test
         for (int i = 0; i < 100000; i++) {
             token_bucket->consumeWithBorrowAndWait(
                 1, throttle_limit / partition_count * 0.8, throttle_limit / partition_count * 1.0);
-            if (!cntl->get_token(1)) {
+            cntl->get_token(1);
```
Comment (levy5307) [inline src/utils/test/token_bucket_throttling_controller_test.cpp:113]: why did you add `cntl->get_token(1)` here?

Comment (Smityz) [reply src/utils/test/token_bucket_throttling_controller_test.cpp:113]: I just split one function to two functions

Comment (levy5307) [reply src/utils/test/token_bucket_throttling_controller_test.cpp:113]: I know, but didn't the function name change to cosume_token?

Comment (Smityz) [reply src/utils/test/token_bucket_throttling_controller_test.cpp:113]: Fix later

File: include/dsn/utils/token_bucket_throttling_controller.h [hunk 2]
```diff
@@ -45,7 +45,10 @@ class token_bucket_throttling_controller
     // return ture means you can get token
     // return false means the bucket is already empty, but the token is borrowed from future.
     // non-blocking
-    bool get_token(int32_t request_units);
+    bool consume_token(int32_t request_units);
+
+    // if the bucket has no tokens, return false
+    bool available();
```
Comment (levy5307) [inline include/dsn/utils/token_bucket_throttling_controller.h:51]:
```
bool available() const;
```

**Review by levy5307 — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]
